src/test/perl/path.t

```diff
@@ -108,9 +108,9 @@ sub verify_exception
     } else {
         ok(! $ec_check->error(), "Error reset after $msg");
     };
-    if ($noreset) {
+    if ($noreset && $mc->{fail}) {
```

Comment:
it changes the logic of the test, from one condition to two. if `$mc->{fail}` is not defined, something is wrong and test will fail, as it is supposed to.